src/slave/slave.cpp

```diff
@@ -3654,6 +3655,34 @@ void Slave::launchExecutor(
       authenticationToken.get(),
       framework->info.checkpoint());
 
+
+  if(environment.find("CRITEO_CNI") != environment.end() && environment["CRITEO_CNI"] == "enabled"){
+
+    // Slave has CNI enabled but doesn't define any networks, public and private subnets must be injected.
+    ContainerInfo* containerInfo = containerConfig.mutable_container_info();
+    containerInfo->set_type(ContainerInfo_Type::ContainerInfo_Type_MESOS);
```

Comment:
I think this type is always MESOS at Criteo; why would we need to change/set it?

Reply:
mutable_container_info() creates a containerInfo object if none already exists (which is usually the case for marathon apps defined using the UI), we need to set the container type because it's a mandatory field